File: customer/src/dtos/externalEvents/customer-was-updated.ts
export interface CustomerWasUpdated {
    id?: string;
    documentNumber?: string;
    name?: string;
    motherName?: string;
    birthDate?: google.protobuf.Timestamp;
    created?: google.protobuf.Timestamp;
    status?: CustomerStatusType;
    updated?: google.protobuf.Timestamp;
    address?: Address;
    contacts?: Contract[];
}
export interface Address {
    street?: string;
    number?: string;
    neighborhood?: string;
    zipCode?: string;
    city?: string;
    state?: string;
    complement?: string;
}
export interface Contract {
    value?: string;
    type?: ContactType;
}
export enum ContactType {
    Email = 0,
    Phone = 1,
}
export enum CustomerStatusType {
    None = 0,
    Simple = 1,
    Partial = 2,
    Complete = 3,
}
export namespace google {
    export namespace protobuf {
        export interface Timestamp {
            seconds?: number;
            nanos?: number;
        }
    }
}

File: antifraud/Antifraud/ExternalContracts/CustomerWasUpdated.cs
using System;
using System.Collections.Generic;

namespace Antifraud.ExternalContracts
{
    public class CustomerWasUpdated
    {
        public string Id { get; set; }
        public string DocumentNumber { get; set; }
        public string Name { get; set; }
        public string MotherName { get; set; }
        public DateTime BirthDate { get; set; }
        public DateTime Created { get; set; }
        public CustomerStatusType Status { get; set;}
        public DateTime Updated { get; set; }
        public Address Address { get; set; }
        public List<Contract> Contacts { get; set; }
    }

    public class Address
    {
        public string Street { get; set; }
        public string Number { get; set; }
        public string Neighborhood  { get; set; }
        public string ZipCode  { get; set; }
        public string City { get; set; }
        public string State { get; set; }
        public string Complement { get; set; }
    }

    public class Contract
    {
        public string Value { get; set; }
        public ContactType Type { get; set; }
    }
}